[LoongArch] Implement LoongArchRegisterInfo::canRealignStack()

wangleiat opened the pull request:
This patch fixes the crash issue in the test:
CodeGen/LoongArch/can-not-realign-stack.ll

Register allocator may spill virtual registers to the stack, which    
introduces stack alignment requirements (when the size of spilled     
    registers exceeds the default alignment size of the stack). If a  
function does not have stack alignment requirements before register   
allocation, registers used for stack alignment will not be preserved. 

Therefore, we should implement `canRealignStack()` to inform the      
register allocator whether it is allowed to perform stack realignment 
operations.

wangleiat commented:
@xry111 @xen0n

SixWeining reviewed (approved):
LGTM.

xen0n reviewed (approved):
If it's based on some other target's similar treatment it could be useful to refer to that arch, otherwise it LGTM as-is.

wangleiat committed:
[LoongArch] Implement LoongArchRegisterInfo::canRealignStack()

This patch fixes the crash issue in the test:
CodeGen/LoongArch/can-not-realign-stack.ll

Register allocator may spill virtual registers to the stack, which
introduces stack alignment requirements (when the size of spilled
    registers exceeds the default alignment size of the stack). If a
function does not have stack alignment requirements before register
allocation, registers used for stack alignment will not be preserved.

Therefore, we should implement `canRealignStack()` to inform the
register allocator whether it is allowed to perform stack realignment
operations.
